psydac/feec/global_projectors.py

```diff
@@ -147,6 +149,9 @@ def __init__(self, space, nquads = None):
                     if quad_x[j] is None:
                         u, w = uw[j]
                         global_quad_x, global_quad_w = quadrature_grid(V.histopolation_grid, u, w)
+                        #"roll" back points to the interval to ensure that the quadrature points are
+                        #in the domain. Probably only usefull on periodic cases
+                        roll_edges(V.domain, global_quad_x) 
```

Comment:
is this related to the H1vec space, or fixing a pre-existing bug ?

Reply:
Can you please further explain what is happening here? `roll_edges` is only relevant in the periodic case, why do you think it is necessary now?

Reply:
@vcarlier Have you answered my question somewhere else?

Reply:
Not yet I was doing tests to be sure that you need it and this is indeed the case : 
In the non periodic case roll edges does nothing so you don't care applying it all the time.
In the periodic case, if you don't use it you might have quadrature point outside of the domain, which might cause problem when you want to define your function by it's value on the domain but it's not periodic on R (for example `x*(x-1)` define a continuous function when restricted to (0,1) with periodic BC, but it's not a periodic function on R so if you don't roll back your going to evaluate it outside of your domain and have wrong values, see `test_derham_projector_2d_hdiv_2` in `test_global_projectors.py` )